### ISSUE TITLE
Column name gets reflected in both columns for models having the same table joined with itself

### ISSUE THREAD

**Issue by Tony-metabase** [metabase/metabase#32499]
### Describe the bug

When you join two tables together inside a model the moment you change any column of one table the change is reflected in the other table

### To Reproduce

1. Go to New Model -> Use Notebook Editor -> Sample DB -> Orders (keep only ID & User ID) JOIN with Orders (keep only User ID) on ID = ID
2. Run it

<img width="1250" alt="image" src="https://github.com/metabase/metabase/assets/110378427/ffd380f5-2ca5-4448-9da6-4d779e296ff7">

3. Go to Model Metadata -> Try to click on the joined User ID and it doesn't work
4. Click on the first table User ID -> Change the name -> it gets reflected in both columns

<img width="1512" alt="image" src="https://github.com/metabase/metabase/assets/110378427/5df4ad0e-3936-4806-aa32-8e773a56ce16">

This doesn't happen if instead of a model you create a question.

### Expected behavior

Able to change both column names individually

### Information about your Metabase installation

1.46.6 and master

### Severity

minor since you can use a question

**Comment by notrom** [comment]
@Tony-metabase I think this happens with questions too, not just models, so your workaround isn't effective. I've been cursing this each time I strike it for years #12685. I would love to see this resolved.

You can use a SQL question but that kinda feels to me like it's going against the spirit of the system.

**Comment by Tony-metabase** [comment]
Hmmm I was not able to hit the GUI question issue. For example i create a similar setup as the above but for GUI:

<img width="1504" alt="image" src="https://github.com/metabase/metabase/assets/110378427/e7cbdd0c-adea-46c5-8c5b-e39d4d04b609">

Then i click on column formatting:

<img width="1510" alt="image" src="https://github.com/metabase/metabase/assets/110378427/0d4feb68-2c0b-4e54-abfd-7270ff4855fa">

Then I update it to test

<img width="1512" alt="image" src="https://github.com/metabase/metabase/assets/110378427/4d557be3-5e63-4d6b-8d9c-ef85f9e26dd2">

Are you having a different result?

**Comment by notrom** [comment]
@Tony-metabase I just tried this in my current prod v0.46.4 and I can still reproduce there:
![image](https://github.com/metabase/metabase/assets/4504437/d0527350-f4ac-4ccf-a564-04565d46dfa3)

But **in v0.47.0-RC1 I can't reproduce it anymore**:
![image](https://github.com/metabase/metabase/assets/4504437/35ccc2a5-705f-4de2-b9a4-cd88298177f3)

Really happy to see that one (#12685) has been sorted! Clearly not the same as this issue, but thanks for prompting me to re-check.

**Comment by Tony-metabase** [comment]
You are right I was testing your issue on the 1.47 just saw the above issue on x.46.6 as well! Thanks for sharing that information

**Comment by shoeferlin** [comment]
Can confirm this still happens. Super annoying and prohibits doing some important analysis.

**Comment by kamilmielnik** [comment]
Reproducible in 0.46.6 and in `master` ([d35c779](https://github.com/metabase/metabase/commit/d35c779976ae6bd79369e9cc6bec8540908f37d2)).

**Comment by darksciencebase** [comment]
This one is also probably a symptom of https://github.com/metabase/metabase/issues/36185

**Comment by kamilmielnik** [comment]
Step 4 is reproducible in master at ef9fb05784. Added to the umbrella issue: #36185.
Cannot reproduce step 3.

**Comment by kamilmielnik** [comment]
This is a field refs issue, a project that is being worked on by the Querying team - I'm relabeling it as such.
We don't have a repro for this issue so we should not close it just yet - reopening.
